File: python/prng/xorshift32.py
#!/usr/bin/env python

## XORSHIFT32 (algorithm by George Marsaglia)
## MZvk @ 2020

import sys

def xorshift32(state):
   state ^= state << 13
   state ^= state << 17
   state ^= state << 5
   return state & 0xffffffff

if __name__ == '__main__':
   try:
      seed = int(sys.argv[1]) & 0xffffffff
      print xorshift32(seed)
   except:
      print "Input must be a single number"

File: perl/prng/xorshift32.pl
#!/usr/bin/env perl

## XORSHIFT32 (algorithm by George Marsaglia)
## Mzvk 2020

use strict;
use warnings;

die "No seed value.\n" unless @ARGV > 0;
print get_num($ARGV[0])."\n";

sub get_num {
   my $seed = shift;
   die "Seed value must be an integer.\n" unless $seed =~ m/^[0-9]+$/;
   $seed &= 0xFFFFFFFF;
   $seed ^= $seed << 13;
   $seed ^= $seed << 17;
   $seed ^= $seed << 5;
   return $seed & 0xFFFFFFFF;
}


